### PR TITLE
16 feature request calculation status logger

### PR DESCRIPTION
**Pull request by kimrojas**
New feature:
1. Created the Logger class. This can be used for the Status logger. You now have a file called STATUS.log in each directory. In addition, it is coded in a way that there is no printing buffer (usually, the print command is printed out all at the end of the run).

Polished feature:
1. Jobscript maker was recoded into a class (used as is)
2. CalculationExecutor class was created to simplify the python handler. Currently it has the `read_jobdict` and `read_args` module.
3. `Jobdict` index fixes, typos etc. 
4. Yaml prefix was initially `atom_run_` with a trailing `_`. This looks bad so I edited the code to automatically add trailing `_`  and now the prefix just needs to be `atom_run`

Current testing info:
1. It has passed the pytest
2. Manual testing on the `single_atom` complete
3. Python jobscript testing was completed successfully by running the calculation itself. I reduced vacuum to 10 A only for faster testing. (ONLY FOR STATE, QE is untested)

### CODE REVIEW

**Comment by kimrojas** [comment]
@hhlim12 @uedar May I ask for a review?

**Comment by kimrojas** [comment]
@jdzla The job dictionary module was recoded here. Please do check if it will be compatible with the csv list reader you are making

**Comment by uedar** [comment]
I also tested it and it worked as expected in STATE 👍  

However, the QE test failed. (./pw.x not found error)
QEbatch does not implement the `link` function to create a pw.x symbolic link in each directory, and `pw_loc` and `pw_name` in input.yaml does not seem to be used.
Do you think we need to implement the link function in QE as well?

File: statebatch/statebatch/STATEbatch.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def link():
         manage_system_params()
         for idx, atom_to_run in enumerate(self.atoms_to_run.values(), start=1):
             cwd = os.getcwd()
-            dirname = self.comp_spec.get('prefix')+str('{:04d}'.format(idx))
+            dirname = self.comp_spec.get('prefix')+"_"+str('{:04d}'.format(idx))
```
Comment (uedar) [inline statebatch/statebatch/STATEbatch.py:65]: Is it possible to implement the same for QE?

Comment (kimrojas) [reply statebatch/statebatch/STATEbatch.py:65]: > However, the QE test failed in my environment.. (./pw.x not found error)
> QEbatch does not implement the `link` function to create a pw.x symbolic link in each directory, and `pw_loc` and `pw_name` in input.yaml does not seem to be used.
> Do you think we need to implement the link function in QE as well?

Yes this is an issue I'm not yet sure how to solve yet. This is actually more of a design problem. Usually in QE, we just use PATH to find the pw.x, in STATE it get's linked (though now I think there is already a way to use PATH). Shall we implement link to QE or path-mode to STATE ?




> Is it possible to implement the same for QE?

Yes. I forgot to apply to QEbatch...

Comment (uedar) [reply statebatch/statebatch/STATEbatch.py:65]: This issue has nothing to do with the implementation of this pull request, so shall we discuss and implement it in a different issue?(https://github.com/hhlim12/STATE-BATCH/issues/52)

Comment (kimrojas) [reply statebatch/statebatch/STATEbatch.py:65]: @uedar trailing "_" is now also applied to QEbatch.

**Review by uedar — APPROVED** [review]
I cleaned up the code based on flake8. Could you please check?

**Comment by uedar** [comment]
I use the vscode extension black for automatic formatting.

**Pull request merged**